### PR TITLE
PSR-16

### PR DESCRIPTION
**Pull request by michaelcullum**

### CODE REVIEW

**Comment by michaelcullum** [comment]
@php-fig/secretaries or @php-fig/psr-15 Please a-okay this for a merge (and do so, you should all have access)

**Comment by dragoonis** [comment]
@michaelcullum updated this PR, rebased on top of latest master.

File: proposed/simplecache-meta.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+3. Scope
+--------
+
+### 3.1 Goals
+
+* A simple interface for basic cache operations.
+* Basic support for operations on multiple keys for performance (round-trip-time)
+  reasons.
+* Support for atomic counters which are essential to implement security features
+  like login-attempt counters and overall counting things under high load.
```
Comment (samdark) [inline proposed/simplecache-meta.md]: "counting things under high load" may be elaborated as "counting things in high concurrency environment"

File: proposed/simplecache.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+PSR-6 solves this problem already, but in a rather formal and verbose way for
+what the most simple use cases need. This simpler approach aims to build a
+standardized layer of simplicity on top of the existing PSR-6 interfaces.
+
+
+### 1.2 Definitions
+
+Definitions for Calling Library, Implementing Library, TTL, Expiration and Key
+are copied from PSR-6 as the same assumptions are true.
```
Comment (samdark) [inline proposed/simplecache.md]: I think it belongs to meta.

File: proposed/simplecache.md [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+
+## 1.1 Introduction
+
+Caching is a common way to improve the performance of any project, making
+caching libraries one of the most common features of many frameworks and
+libraries. Interoperability at this level means libraries can drop their
+own caching implementations and easily rely on the one given to them by the
+framework, or another dedicated cache library.
+
+PSR-6 solves this problem already, but in a rather formal and verbose way for
```
Comment (samdark) [inline proposed/simplecache.md]: It could be a good idea to move it to meta.

File: proposed/simplecache.md [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+but MUST treat an item as expired once its Expiration Time is reached. If a calling
+library asks for an item to be saved but does not specify an expiration time, or
+specifies a null expiration time or TTL, an Implementing Library MAY use a configured
+default duration. If no default duration has been set, the Implementing Library
+MUST interpret that as a request to cache the item forever, or for as long as the
+underlying implementation supports.
+
+*    **Key** - A string of at least one character that uniquely identifies a
+cached item. Implementing libraries MUST support keys consisting of the
+characters `A-Z`, `a-z`, `0-9`, `_`, and `.` in any order in UTF-8 encoding and a
```
Comment (samdark) [inline proposed/simplecache.md]: Why `-` is excluded?

File: proposed/simplecache.md [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+length of up to 64 characters. Implementing libraries MAY support additional
+characters and encodings or longer lengths, but must support at least that
+minimum.  Libraries are responsible for their own escaping of key strings
+as appropriate, but MUST be able to return the original unmodified key string.
+The following characters are reserved for future extensions and MUST NOT be
```
Comment (samdark) [inline proposed/simplecache.md]: It's common to `json_encode` or `md5` structures to get their keys. That would prevent such use. Is it intended?

File: proposed/simplecache.md [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+namespace Psr\SimpleCache;
+
+interface CacheInterface
+{
+    const TTL_MINUTE = 60;
+    const TTL_HOUR = 3600;
+    const TTL_DAY = 86400;
+
+    /**
+     * Fetched a value from the cache.
```
Comment (samdark) [inline proposed/simplecache.md]: Fetched → Fetch?

File: proposed/simplecache.md [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+     * @return bool True on success and false on failure
+     */
+    public function set($key, $value, $ttl = null);
+
+    /**
+     * Delete an item from the cache by its unique key
+     *
+     * @param string $key The unique cache key of the item to delete
+     *
+     * @return void
```
Comment (samdark) [inline proposed/simplecache.md]: Shouldn't boolean be returned?

File: proposed/simplecache.md [hunk 8]
```diff
@@ -0,0 +1,230 @@
...
+     */
+    public function delete($key);
+
+    /**
+     * Wipe clean the entire cache's keys
+     *
+     * @return void
```
Comment (samdark) [inline proposed/simplecache.md]: Shouldn't boolean be returned?

File: proposed/simplecache.md [hunk 9]
```diff
@@ -0,0 +1,230 @@
...
+     * @param array|Traversable     $items An array of key => value pairs for a multiple-set operation.
+     * @param null|int|DateInterval $ttl   Optional. The amount of seconds from the current time that the item will exist in the cache for.
+     *                                     If this is null then the cache backend will fall back to its own default behaviour.
+     *
+     * @return bool True on success and false on failure
+     */
+    public function setMultiple($items, $ttl = null);
+
+    /**
+     * Delete multiple cache items in a single operation
```
Comment (samdark) [inline proposed/simplecache.md]: Should mention what to do in case of non existing keys.

**Comment by michaelcullum** [comment]
This PR really does need merging so it's in the repo as proposed. @dragoonis Could you fix the above issues then feel free to hit the green button. :)

File: proposed/simplecache-meta.md (hunk 1 above)
Comment (dragoonis) [inline proposed/simplecache-meta.md]: fixed.

File: proposed/simplecache.md (hunk 6 above)
Comment (dragoonis) [inline proposed/simplecache.md]: Fetch. Fixed.

**Comment by michaelcullum** [comment]
Awesome. Further things can be discussed on the ML?

**Comment by dragoonis** [comment]
@michaelcullum I can't hit the green button. After many years of contributing my write access has been removed from fig-standards and the php-fig.org website repositories.

**Comment by dragoonis** [comment]
@michaelcullum yes things can be discussed on the ML

**Pull request merged**